src/code/z_actor.c

```diff
@@ -1873,10 +1873,32 @@ s32 Actor_ProcessTalkRequest(Actor* actor, GameState* gameState) {
     return false;
 }
 
-// Actor_OfferTalk / Actor_OfferGetItemExchange? Seems to be called with PLAYER_IA_MINUS1 if the same actor used
-// Actor_OfferGetItem.
-// This function is also used to toggle the "Speak" action on the A button
-s32 func_800B8500(Actor* actor, PlayState* play, f32 xzRange, f32 yRange, PlayerItemAction exchangeItemId) {
+/**
+ * This function covers various interactions with the player actor, using Exchange Items (see `ItemAction` enum).
+ * It is typically used to exchange items with the player without first speaking, but also has other purposes.
+ *
+ * This function carries an exchange item request to the player actor if context allows it (e.g. the player is in range
+ * and not busy with certain things). The player actor performs the requested action itself.
+ *
+ * The following description of what the `ItemAction` values can do is provided here for completeness, but these
+ * behaviors are entirely out of the scope of this function. All behavior is defined by the player actor.
+ *
+ * - Positive values in range (`PLAYER_IA_NONE < ItemAction < PLAYER_IA_MAX`):
+ *    Offers the ability to exhange an item with the player.
+ *    Not all postivie values are implemented properly for this to work.
+ *    Working ones seem to be PLAYER_IA_PICTO_BOX and PLAYER_IA_BOTTLE_MIN <= ItemAction < PLAYER_IA_MASK_MIN
+ * - `PLAYER_IA_HELD`:
+ *    Used by actors/player to continue the current conversation.
+ * - `PLAYER_IA_NONE`:
+ *    Allows the player to speak to the actor (by pressing A).
+
+ *
+ * @return true If the player actor is capable of accepting the offer.
+ *
+ * Note: There is only one instance of using this for actually exchanging an item with the player
+ * without speaking first. Every other instance is to either offer to speak, or continue the current conversation.
+ */
+s32 Actor_OfferExchangeItem(Actor* actor, PlayState* play, f32 xzRange, f32 yRange, PlayerItemAction exchangeItemId) {
```

Comment:
Thanks for these docs! I've actually been interested in the exchange system since it was interlinked with bottles which I was looking at in detail, but then the exchange system is also interlinked with the talk system.

I think anytime you want to talk to an actor, you need to go through this function. An actor needs to indicate that it can be talked to through an offer, and I think this function name should capture that. Of course, they mainly do that through the wrapper of the function `Actor_OfferSpeak`, but this function should capture that you can talk with the choice of exchange.

My rough first idea for these function names are:
`Actor_OfferToTalkWithExchange`
`Actor_OfferToTalkWithExchangeRadius`
`Actor_OfferToTalk`
`Actor_OfferToTalkNearby`

I also think `Talk` should be used with `Speak` because this would apply to both `Speak` and `Check` (for example, signs and posters in Clock Town).

It's definitely open to discussion though, thoughts?

Reply:
I had actually been considering something like that. Though I would probably suggest dropping the `To` and `With` as I'm not sure they add much. Something like
* `Actor_OfferTalkExchange`
* `Actor_OfferTalkExchangeRadius`
* `Actor_OfferTalk`
* `Actor_OfferTalkNearby`
And for completeness the Subs Functions
* `SubS_OfferItem` -> `SubS_OfferTalkExchange` (while getItem can be used here, the focus seemed to be on the exchange system so I think this is fine)
* `SubS_OfferExchangeItemFacing` -> `SubS_OfferTalkExchangeFacing`